woocommerce/woocommerce-gutenberg-products-block
# Ensure the content of the patterns is also AI-generated

## What

Update all patterns to ensure all the texts are populated with AI-generated content. The text should dynamically change depending on what the user decides to add as a description for their business on their AI settings screen and also during the upcoming onboarding screen.

## Why

-

## Testing Instructions



_Please consider any edge cases this change may have, and also other areas of the product this may impact._

1. Create a new [JN install](https://jurassic.ninja/) with both Jetpack and Woocommerce installed and activated.
3. Install and activate WooCommerce Blocks from this zip file:  [woocommerce-gutenberg-products-block.zip](https://wcblocks.wpcomstaging.com/wp-content/uploads/woocommerce-gutenberg-products-block-10997.zip)

5. In your wp-admin, you should see the following section for Jetpack:

<img width="1803" alt="Screenshot 2023-09-11 at 18 55 38" src="https://github.com/woocommerce/woocommerce-blocks/assets/15730971/fc804759-f724-4fa2-829d-7bfd441962ec">

6. Click on "Set up Jetpack" and make sure the connection is successful:

<img width="1994" alt="Screenshot 2023-09-11 at 18 58 30" src="https://github.com/woocommerce/woocommerce-blocks/assets/15730971/661ee4c5-a0ac-40fd-9e5e-835a0e4c1cb7">

7. From your admin screen, click on Tools > Plugin editor: 

<img width="337" alt="Screenshot 2023-09-11 at 18 34 23" src="https://github.com/woocommerce/woocommerce-blocks/assets/15730971/c353bd13-bf00-4bc8-93d6-d827ca99997f">

8. Select the WooCommerce Blocks plugin and add the following within your woocommerce-blocks/woocommerce-gutenberg-products-block.php file and save:

```php
function connect_to_endpoint() {
	update_option( 'woocommerce_blocks_allow_ai_connection', true );

	$request = new WP_REST_Request( 'POST', '/wc/store/patterns' );

	$request->set_param( 'business_description', 'A store that sells organic food' );

	$response = rest_do_request( $request )->get_data();

	error_log( print_r( $response, true ) );
}

add_action( 'admin_init', 'connect_to_endpoint' );
```

9. Create a new post
10. Insert the `Featured Category Triple` pattern (or any other woocommerce-blocks pattern).
11. Make sure the images on the pattern match the business description you provided: 'A store that sells organic food'
12. Go back to the plugin editor and update the business description over there to something else (e.g. books, clothes, sports gear etc.) and save
13. Return to your post editor, save it, and refresh the page.
14. Insert the same `Featured Category Triple` pattern again
15. Notice how all images and text were updated to now match what you selected as the new description for your store


## Screenshots or screencast



| Before | After |
| ------ | ----- |
|        |       |

## WooCommerce Visibility


Required:


## Checklist

Required:

Conditional:

## Changelog


N/A
